### PR TITLE
Adding CentOS Installation

### PR DESCRIPTION
**Pull request by peasead**
Added a complete and turnkey installation for CentOS 7.4.

### CODE REVIEW

**Comment by peasead** [comment]
Is there a way to sign my commit after the fact?

File: docs/guides/centos-installation.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+# Set your IP address as a variable. This is for instructions below.
+IP="$(hostname -I | sed -e 's/[[:space:]]*$//')"
+
+# Install dependencies
+sudo yum install epel-release -y
+sudo yum install mariadb-server npm gcc-c++ git bzip2 -y
+
+# Stage HackMD for building
+sudo npm install -g uws node-gyp tap webpack grunt yarn
```
Comment (SISheogorath) [inline docs/guides/centos-installation.sh]: You shouldn't install node packages, that are not managed by yum, globally.

File: docs/guides/centos-installation.sh [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+sudo yarn add -D webpack-cli
+sudo git clone https://github.com/hackmdio/hackmd.git /opt/hackmd/
+cd /opt/hackmd
```
Comment (SISheogorath) [inline docs/guides/centos-installation.sh]: When you create the directory and chown it to the current user before you clone it, you can minimize the attack surface.

```
sudo mkdir /opt/hackmd
sudo chown $(id -u):$(id -g) /opt/hackmd
git clone https://github.com/hackmdio/hackmd.git /opt/hackmd
cd  /opt/hackmd
bin/setup
```

File: docs/guides/centos-installation.sh [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+# Update the HackMD configuration files
+sudo sed -i 's/"username":\ ""/"username":\ "hackmd"/' /opt/hackmd/config.json
+sudo sed -i 's/"password":\ ""/"password":\ "'$hackmdpassphrase'"/' /opt/hackmd/config.json
+sudo sed -i 's/5432/3306/' /opt/hackmd/config.json
+sudo sed -i 's/postgres/mysql/' /opt/hackmd/config.json
+sudo sed -i 's/change\ this/mysql:\/\/hackmd:'$hackmdpassphrase'@localhost:3306\/hackmd/' /opt/hackmd/.sequelizerc
+
+# Build HackMD
+sudo npm run build --prefix /opt/hackmd/
```
Comment (SISheogorath) [inline docs/guides/centos-installation.sh]: When done above this works also without sudo and prevents malicious submodules from hurting your system

File: README.md [hunk 4]
```diff
@@ -25,6 +25,7 @@ Thanks for using! :smile:
   - [Getting started (Native install)](#getting-started-native-install)
     - [Prerequisite](#prerequisite)
     - [Instructions](#instructions)
+      - [CentOS Installation](docs/guides/centos-installation.sh)
```
Comment (SISheogorath) [inline README.md]: You need to add a section into this document as it's a ToC. Please refer to the Docker section.

File: docs/guides/centos-installation.sh [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+sudo sh -c 'echo [mysqld] > /etc/my.cnf.d/bind-address.cnf'
+sudo sh -c 'echo bind-address=127.0.0.1 >> /etc/my.cnf.d/bind-address.cnf'
+sudo systemctl restart mariadb.service
+mysql_secure_installation
+
+###############################
+### Clear your Bash history ###
+###############################
+# We don't want anyone snooping around and seeing any passphrases you set
+cat /dev/null > ~/.bash_history && history -c
```
Comment (SISheogorath) [inline docs/guides/centos-installation.sh]: I'm not sure this is perfect as people may want to keep their history.

**Comment by SISheogorath** [comment]
If you make it to fix the request changes and signoff your commits, feel free to reopen.

**Pull request closed without merging**